marketplace-client/src/marketplace/src/app/checkout/checkout.component.html

```diff
@@ -0,0 +1,274 @@
+<div class="container">
+  <div class="mt-3"></div>
+  <div *ngIf="!isAuth() && this.isVisibleBanner">
+    <div class="d-flex p-2 flex-column border rounded shadow">
+      <svg 
+        xmlns="http://www.w3.org/2000/svg" 
+        width="16" 
+        height="16" 
+        fill="currentColor" 
+        class="bi bi-x" 
+        viewBox="0 0 16 16"
+        (click)="hideBanner()">
+        <path d="M4.646 4.646a.5.5 0 0 1 .708 0L8 7.293l2.646-2.647a.5.5 0 0 1 .708.708L8.707 8l2.647 2.646a.5.5 0 0 1-.708.708L8 8.707l-2.646 2.647a.5.5 0 0 1-.708-.708L7.293 8 4.646 5.354a.5.5 0 0 1 0-.708z"/>
+      </svg>
+
+      <h1 class="text-danger font-italic text-center">Looks like you not authorized :(</h1>
+      <div class="row p-2 d-flex justify-content-center">
+        <div class="col-sm-3">
+          <a routerLink="/accounts/login" class="btn btn-secondary btn-md order-1 w-100">Login</a>
+        </div>
+      </div>
+
+      <div class="row p-2 d-flex justify-content-center">
+        <div class="col-sm-3">
+          <a routerLink="/accounts/register" class="btn btn-secondary btn-md order-1 w-100">Register</a>
+        </div>
+      </div>
+    </div>
+  </div>
+  
+
+  <div *ngIf="items.length === 0" class="text-center my-5">
+    <h4>Your Cart Is Empty</h4>
+    <button class="btn btn-primary btn-lg mt-3" routerLink="/products">GO TO SHOP</button>
+  </div>
+
+  <div *ngIf="items.length !== 0">
+
+    <form *ngIf="!submitted" [formGroup]="orderDetailsForm" (ngSubmit)="onSubmit()">
+      <hr />
+      <h1>Checkout form</h1>
+      <hr />
+
+      <div *ngIf="!isAuth()" class="row d-flex p-2 justify-content-center">
```

Comment:
https://medium.com/showpad-engineering/why-you-should-never-use-function-calls-in-angular-template-expressions-e1a50f9c0496